core/src/main/scala/io/finch/Input.scala

```diff
@@ -103,17 +103,16 @@ object Input {
       Input(copied, i.route)
     }
 
-    def apply[F[_]: Effect, S[_[_], _], A](s: S[F, A])(implicit
+    def apply[F[_], S[_[_], _], A](s: S[F, A], dispatcher: Dispatcher[F])(implicit
```

Comment:
This is another place that I'm not in love with but we should leave it for another PR. I don't have a ton of ideas just yet but I'm thinking to extract this and awaitX methods into their own testing package (or syntax?) that's hopefully structured around the existing dispatcher.